izer/assets.py

```diff
@@ -222,36 +226,245 @@ def vscode(
+                content = None
+                with open(template, 'r', encoding="UTF-8") as f:
+                    content = f.read()
+                    content = content.replace("##__TARGET__##", target.upper()). \
+                        replace("##__BOARD__##", board). \
+                        replace("##__PROGRAM_FILE__##", program_file). \
+                        replace("##__SYMBOL_FILE__##", symbol_file). \
+                        replace("##__M4_OCD_INTERFACE_FILE__##", m4_ocd_interface_file). \
+                        replace("##__M4_OCD_TARGET_FILE__##", m4_ocd_target_file). \
+                        replace("##__RV_OCD_INTERFACE_FILE__##", rv_ocd_interface_file). \
+                        replace("##__RV_OCD_TARGET_FILE__##", rv_ocd_target_file). \
+                        replace("\"##__I_PATHS__##\"", i_paths_parsed). \
+                        replace("\"##__DEFINES__##\"", defines_parsed). \
+                        replace("\"##__V_PATHS__##\"", v_paths_parsed). \
+                        replace("##__V_ARM_GCC__##", v_arm_gcc). \
+                        replace("##__V_XPACK_GCC__##", v_xpack_gcc). \
+                        replace("##__OCD_PATH__##", ocd_path). \
+                        replace("##__ARM_GCC_PATH__##", arm_gcc_path). \
+                        replace("##__XPACK_GCC_PATH__##", xpack_gcc_path). \
+                        replace("##__MAKE_PATH__##", make_path)
+
+                write = True
+                if out_file.exists():
+                    if not overwrite or utils.compare_content(content, out_file):
+                        write = False
+
+                if write:
+                    with open(out_file, "w+", encoding="UTF-8") as f:
+                        f.write(content)
+                    os.chmod(out_file, stat.S_IRWXU | stat.S_IRGRP | stat.S_IWGRP | stat.S_IROTH)
+                    if out_file not in updated:
+                        updated.append(out_file)
+
+                    # print(f"Wrote {os.path.basename(out_loc)}")  # Uncomment to debug
 
             else:
                 # There is a non-template file to copy
-                shutil.copy(os.path.join(directory, file), out_path)
-                os.chmod(out_path, stat.S_IRWXU | stat.S_IRGRP | stat.S_IWGRP | stat.S_IROTH)
-                # print(f"Wrote {os.path.basename(file)}") # Uncomment to debug
+                in_file = Path(directory).joinpath(file)
+                out_file = Path(out_path).joinpath(file)
+
+                write = True
+                if out_file.exists():
+                    if not overwrite or (utils.hash_file(in_file) == utils.hash_file(out_file)):
+                        write = False
+
+                if write:
+                    shutil.copy(in_file, out_path)
+                    os.chmod(out_file, stat.S_IRWXU | stat.S_IRGRP | stat.S_IWGRP | stat.S_IROTH)
+                    if out_file not in updated:
```

Comment:
if you make 'updated' a set(), then this condition is not needed. Just use 'updated.add(out_file)'